src/TypeScript/EslintBridgeClient/EslintBridgeKeepAlive.cs

```diff
@@ -0,0 +1,110 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.SystemAbstractions;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+
+namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient
+{
+    /// <summary>
+    /// Component that is responsible for sending keep-alive messages
+    /// to the eslint-bridge
+    /// </summary>
+    internal interface IEslintBridgeKeepAlive : IDisposable
+    {
+    }
+
+    internal sealed class EslintBridgeKeepAlive : IEslintBridgeKeepAlive
+    {
+        // The eslintbridge process will shutdown after 15 seconds.
+        // We send a keep alive every 5 seconds.
+        private const double MillisecondsToWaitBetweenKeepAlives = 5000;
+
+        private readonly IEslintBridgeProcess process;
+        private readonly IEslintBridgeHttpWrapper httpWrapper;
+        private readonly ITimer timer;
+        private readonly ILogger logger;
+
+        public EslintBridgeKeepAlive(IEslintBridgeProcess process, ILogger logger)
+            : this(process, logger, new EslintBridgeHttpWrapper(logger), new TimerWrapper())
+        {
+        }
+
+        internal /* for testing */ EslintBridgeKeepAlive(IEslintBridgeProcess process, ILogger logger,
+            IEslintBridgeHttpWrapper httpWrapper, ITimer timer)
+        {
+            this.process = process;
+            this.logger = logger;
+            this.httpWrapper = httpWrapper;
+
+            this.timer = timer;
+            this.timer.AutoReset = true;
+            this.timer.Interval = MillisecondsToWaitBetweenKeepAlives;
+            this.timer.Elapsed += OnKeepAliveTimerElapsed;
+
+            // Note: we're starting the timer immediately, even though the process might not be running
+            // yet (i.e. the user hasn't tried to analyse a JS/TS file)
+            this.timer.Start();
```

Comment:
This means the timer will start as soon as this class is created i.e. as soon as the `EslintBridgeClient` instance is created.

This happens when the JS/TS analyzer is MEF-imported i.e. before we know whether there are any JS/TS to analyze.

This was the one advantage of the version in #3203 - the timer was not started until an analysis was triggered.

As it is, two timers (one for JS, one for TS) will start when the `AnalyzerController` is created. They will fire in the background every five seconds, and early-out because the process is not running.

I'm not sure this is worth worrying about, though. What do you think?